### DISCUSSION TITLE
Pole figures for custom crystal orientation data

### DISCUSSION THREAD

**Discussion by abarun22** [mtex-toolbox/mtex#1980] [Ask Anything]
Hi,
I am interested in plotting pole figures for an arbitrary set of crystal orientation data and i am figuring out the possibilities existing within Mtex to accomplish that. The easiest way to do this is to store such data in a .uxd format and use Mtex plotting functions to create them. I welcome any suggestions as whether the attached format can be used in any of generic functionalities that Mtex has to plot these pole figures.
Kind regards,
Arun
[pcrystal_standard.txt](https://github.com/mtex-toolbox/mtex/files/12884276/pcrystal_standard.txt)

**Comment by abarun22** [comment]
Hi,
I progressed on this a bit, reading a set of crystal euler angles in the format i described above and use that to plot the pole figures that i am interested in. After plotting each of the crystal orientations, i hold the plot with "hold on;" to make sure the points are plotted in the same figure. One of the plot corresponding to the upper symmetry elements, fail to recognize the hold and does not add the points, while the lower one does it correctly. Attached figure might explain this better.

```
filename = 'pcrystal_standard.csv';
fileID = fopen(filename);
C = textscan(fileID,'%f %f %f');
fclose(fileID);
ncrys = cellfun(@length, C(1,1));
     for i =1:ncrys
        phi1=deg2rad(C{1,1}(i,1));
        phi=deg2rad(C{1,2}(i,1));
        phi2=deg2rad(C{1,3}(i,1));
        ori = orientation.byEuler(phi1,phi,phi2,cs,ss);
        h = Miller({1,0,0},cs);
        r1 = ori * h.symmetrise;        
        plot(r1);
        hold on;
     end
hold off;

```
Not sure why this happens, but probably some one has answers.
![image](https://github.com/mtex-toolbox/mtex/assets/40111479/eda85e74-bde2-428c-af23-9fee6d7c180e)

Kind regards,
Arun

**Comment by kilir** [comment]
Hi,
if you add the option `'add2all'` for subsquent plots after `'hold on'`, points should appear on both plots.

However, if you simply want to plot a polefigure from the data, you can do:
```
cs = crystalSymmetry('1') % whatever it might be
ori = loadOrientation_generic('pcrystal_standard.txt','CS',cs,'ColumnNames',{'Euler1','Euler2','Euler3'})
h = Miller({1,0,0},cs);
plotPDF(ori,h)
```

Hope this helps.

Cheers,
Rüdiger
